docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,299 @@
+* Publishing: An annotated tag `<component>/v<major>.<minor>.<patch>` is the authoritative record of a published artifact.
+
+## SemVer Principles for Root Component
+
+### Normal Case (Automatic)
+
+The `ocm` root component version follows these automated rules for independent sub-component releases:
+
+* **Patch bump (x.y.z → z++)**: triggered by any sub-component patch release
+* **Minor bump (x.y.z → y++)**: triggered by any sub-component feature addition  
+* **Major bump (x.y.z → x++)**: triggered by any sub-component new major release
+
+Every OCM root release represents a reproducible snapshot of all sub-component versions and maintains strict SemVer compliance. The root component version is automatically computed and updated in `/VERSIONS.yaml` when any sub-component publishes a release.
+
+### Edge Case: Coordinated Multi-Component Releases
+
+When multiple sub-components must be released together to solve a single problem:
+
+**Problem:** Automatic root bumping would create multiple OCM releases instead of one logical `ocm`release containing all related changes.
+
+**Solution:** Manual override workflow for coordinated releases:
+
+1. **Sub-component releases with root suppression:**
+
+   Execute the GitHub Action release workflow for each affected sub-component with the "suppress-root-release" flag enabled (flag name to be finalized).
+
+   * Run release workflow for `ocm-cli` with flag: `suppress-root-release = true`
+   * Run release workflow for `ocm-controller` with flag: `suppress-root-release = true`
+   * Each workflow completes its normal release process (tag creation, artifact publishing) but skips the automatic OCM root release trigger.
+
+2. **Manual root release consolidation:**
+
+   After all coordinated sub-component releases are completed, manually trigger the "Create OCM Root Release" workflow:
+
+   * Reads current `/VERSIONS.yaml` state to identify all unreleased changes
+   * Computes appropriate bump type (highest sub-component bump type across all coordinated releases)
+   * Creates single OCM release tag containing all coordinated changes
+
+**Bump calculation for coordinated releases:** The root component bump type is determined by the highest impact change across all coordinated sub-components (e.g., if one patch + one minor → minor bump for root).
+
+## Considered Options
+
+I explicitly considered the following options before making the decision above. Recording them here improves traceability.
+
+### Versioning options
+
+* Per-component `VERSION` file: each component contains a `VERSION` file in its directory that is the working state for that component.
+* Central `VERSIONS.yaml`: a single root `VERSIONS.yaml` that is the canonical snapshot of component versions (human-edited or bot-maintained).
+* Hybrid: per-component `VERSION` files for working state, plus a generated `/VERSIONS.yaml` for published snapshots.
```

Comment:
I dont think this makes sense. the generated VERSIONS.yaml would be unnecessary if it would always contain the latest versions. The only time where a dedicated file would make sense is if you cannot generate it

Reply:
as mentioned above, when changing the proposal and implementing using tags, we can use code to calculate all that on-the-fly.